src/Findologic/Request/Handler/FilterHandler.php

```diff
@@ -268,4 +271,22 @@ private function parseFindologicFiltersForShopware(FiltersExtension $filterExten
 
         return array_merge($actualResult, $result);
     }
+
+    /**
+     * @param FilterValue[] $values
+     */
+    private function handleCategoryFilters(array $values, array &$filterValues): void
```

Comment:
I really dislike passing values per reference, if it doesn't serve an immediate purpose, other than making it easier accessible. Try to return the value instead.

Reply:
This would require using `array_merge` inside the loop (due to recursive nature of the method), which is not very efficient in my opinion.

Reply:
Sure makes sense then.